### PR TITLE
ppswatch: Add useful stats output.

### PR DESCRIPTION
**Pull request by ImmortalGuardian**
Calculate mean value and standard deviation of captured offsets. Display them at the exit.

Signed-off-by: Andrey Drobyshev <immortalguardian1@gmail.com>

### CODE REVIEW

File: ppswatch.c [hunk 1]
```diff
@@ -252,6 +282,12 @@ int main(int argc, char *argv[])
 	if (margin)
 		printf("using margin %d\n", margin);
 
+	offsets = (long*)calloc(bufsize, sizeof(long));
```
Comment (gerasiov) [inline ppswatch.c]: offsets = calloc(bufsize, sizeof(*offsets));

File: ppswatch.c [hunk 2]
```diff
@@ -130,8 +135,20 @@ int fetch_source(pps_handle_t handle, int avail_mode)
 		ts.tv_nsec -= NSEC_PER_SEC;
 	}
 
-	total++;
 	div = ts.tv_nsec;
+
+	if (bufsize < total) {
+		bufsize *= 2;
+		offsets = (long*)realloc(offsets, bufsize);
```
Comment (gerasiov) [inline ppswatch.c]: offsets = realloc(offsets, bufsize*sizeof(*offsets));

**Comment by gerasiov** [comment]
Andrey, please splash both commits into one.

**Comment by ImmortalGuardian** [comment]
@gerasiov, done.

**Comment by ago** [comment]
Hi! What's this for and why you used this algorithm for variance calculation? Why not an online algorithm, for example?

**Comment by gerasiov** [comment]
Yep. That was the next question. =)
Endless (until allocation failure occurs) buffer enlargement could be bad is someone leave the tool run for days.

**Comment by ImmortalGuardian** [comment]
@ago, well, since this tool is mostly for measuring accuracy rate, I suppose it is reasonable to obtain some more numerical characteristics. When correction is stable enough, maximum divergence becomes less informative.
And yes, you're inarguably right: online algorithm is much better, thanks:) I fixed it.

**Comment by ago** [comment]
Good, thanks :)

File: ppswatch.c [hunk 3]
```diff
@@ -132,6 +137,9 @@ int fetch_source(pps_handle_t handle, int avail_mode)
 
 	total++;
 	div = ts.tv_nsec;
+	delta = div - mean;
+	mean += delta / total;
+	variance = (total - 1) * variance / total + (total - 1) * sqr(delta) / sqr(total);
```
Comment (gerasiov) [inline ppswatch.c]: Why not to use Welford variance algorithm?

**Comment by ImmortalGuardian** [comment]
Upd: Welford algorithm for variance calculation is used now.

**Pull request closed without merging**